src/coreclr/jit/lsra.cpp

```diff
@@ -8130,10 +8168,12 @@ void LinearScan::resolveEdges()
 // resolveEdge: Perform the specified type of resolution between two blocks.
 //
 // Arguments:
-//    fromBlock     - the block from which the edge originates
-//    toBlock       - the block at which the edge terminates
-//    resolveType   - the type of resolution to be performed
-//    liveSet       - the set of tracked lclVar indices which may require resolution
+//    fromBlock              - the block from which the edge originates
+//    toBlock                - the block at which the edge terminates
+//    resolveType            - the type of resolution to be performed
+//    liveSet                - the set of tracked lclVar indices which may require resolution
+//    terminatorConsumedRegs - the registers consumed by the terminator node.
+//                             These registers will be used after any resolution added to 'fromBlock'.
```

Comment:
```suggestion
//                             These registers will be used after any resolution added at the end of the 'fromBlock'.
```

Probably you meant this. But not sure if I fully understand this. Inside `getTempRegForResolution0` we say that they are not free to be used as temp registers, but why is that the case? They are consumed already and should be fine to use them, right?

Reply:
Yes, I meant that. Will update.

>  They are consumed already and should be fine to use them, right?

No, resolution nodes are added before the terminating node and any registers it uses. The consumed registers are computed here:
https://github.com/dotnet/runtime/blob/f1bdd5a6182f43f3928b389b03f7bc26f826c8bc/src/coreclr/jit/lsra.cpp#L7655-L7726

Reply:
> resolution nodes are added before the terminating node and any registers it uses.

Of course, that's right. I was thinking something else. But then how did it work so far? Wasn't this problem always?

Reply:
I'm not totally sure how it worked previously either. If I had to guess the problematic cases (i.e. switches/JCMP blocks) always end up doing resolution in the target blocks (except for the shared critical edges cases, which was skipped before). I think that would make sense since these kinds of blocks should always have 2 or more successors.

Reply:
That could be the case. So were you hitting assert or something when you added support for shared critical edges and that made you consider the `terminatorConsumeRegs`?

Reply:
No, it was a Fuzzlyn test case, here was the codegen for that case: https://github.com/dotnet/runtime/pull/81216#issuecomment-1409363747

Reply:
> No, it was a Fuzzlyn test case, here was the codegen for that case: [#81216 (comment)](https://github.com/dotnet/runtime/pull/81216#issuecomment-1409363747)

Ah, I missed that comment and findings. That makes sense then.

Reply:
Can you add a test case for that as well?

Reply:
The test case is massive, I think weekly Antigen/Fuzzlyn runs should be ok to cover it.